src/Features/CSharp/Portable/DocumentationComments/CodeFixes/CSharpAddDocCommentNodesCodeFixProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.DiagnosticComments.CodeFixes
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.AddDocCommentNodes), Shared]
+    [ExtensionOrder(After = PredefinedCodeFixProviderNames.ImplementInterface)]
+    internal class CSharpAddDocCommentNodesCodeFixProvider
+        : AbstractAddDocCommentNodesCodeFixProvider<XmlElementSyntax, XmlNameAttributeSyntax, XmlTextSyntax, MemberDeclarationSyntax>
+    {
+        /// <summary>
+        /// Parameter has no matching param tag in XML comment
+        /// </summary>
+        private const string CS1573 = nameof(CS1573);
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(CS1573);
+
+        protected override string NodeName { get; } = "param";
+
+        protected override List<XmlNameAttributeSyntax> GetNameAttributes(XmlElementSyntax node)
+            => node.StartTag.Attributes.OfType<XmlNameAttributeSyntax>().ToList();
+
+        protected override string GetValueFromNameAttribute(XmlNameAttributeSyntax attribute)
+            => attribute.Identifier.Identifier.ValueText;
+
+        protected override SyntaxNode GetDocCommentNode(SyntaxTriviaList leadingTrivia)
+            => leadingTrivia.FirstOrDefault(f => f.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia)).GetStructure();
```

Comment:
any chance this could be a problem if, say the node had multiple DocComment comments on it?  i.e. something strange like:

```
/// ...
//    Note this is two slashes
/// actual doc comemnt with issue
public void Whatever
```